packages/commerce-server/src/prisma-next-serializer.ts

```diff
@@ -8,6 +8,8 @@ export function convertToSerializeForNextResponse(result: any) {
       result[resultKey] = result[resultKey].toISOString()
     } else if (result[resultKey]?.constructor?.name === 'Decimal') {
       result[resultKey] = result[resultKey].toNumber()
+    } else if (result[resultKey] instanceof Object) {
+      result[resultKey] = convertToSerializeForNextResponse(result[resultKey])
     }
```

Comment:
This allows the serializer to work with "nested" prisma objects. Prisma queries that do an include of a related record are going to potentially have things like `datetime` values nested below that will also need to be serialized.

e.g.:

```
// purchase
{
  id: '123',
  productId: '345',
  createdAt: <DateTime Obj>,
  // ...
  bulkCoupon: {
    id: '678',
    createdAt: <DateTimeObj>, // <== this will error unless we serialize recursively
  },
}
```